packages/grid_client/src/primitives/network.ts

```diff
@@ -133,8 +183,24 @@ class Network {
     }
     znet.ip_range = this.ipRange;
     znet.wireguard_private_key = keypair.privateKey;
-    znet.wireguard_listen_port = await this.getFreePort(node_id);
-    znet["node_id"] = node_id;
+    znet.wireguard_listen_port = await this.getFreePort(nodeId);
+    znet["node_id"] = nodeId;
+
+    if (mycelium) {
+      const myceliumNetworkSeed = myceliumSeed.find(item => item.nodeId === nodeId);
+      if (myceliumNetworkSeed?.myceliumSeed) {
+        validateHexSeed(myceliumNetworkSeed.myceliumSeed, 32);
+        znet.mycelium = {
+          hex_key: myceliumNetworkSeed?.myceliumSeed,
+          peers: [],
+        };
+      } else {
+        znet.mycelium = {
+          hex_key: generateRandomHexSeed(32),
+          peers: [],
```

Comment:
you can just generate the seed before this and use the provided seed or the generated seed instead of duplicating the code.